pyresample/geometry.py

```diff
@@ -185,29 +185,33 @@ def get_lonlat(self, row, col):
             raise ValueError('lon/lat values are not defined')
         return self.lons[row, col], self.lats[row, col]
 
-    def get_lonlats(self, data_slice=None, **kwargs):
+    def get_lonlats(self, data_slice=None, chunks=None, **kwargs):
         """Base method for lon lat retrieval with slicing"""
 
-        if self.lons is None or self.lats is None:
+        lons = self.lons
+        lats = self.lats
+        if lons is None or lats is None:
             raise ValueError('lon/lat values are not defined')
-        elif data_slice is None:
-            return self.lons, self.lats
-        else:
-            return self.lons[data_slice], self.lats[data_slice]
+        elif chunks is not None:
+            import dask.array as da
+            if isinstance(lons.data, da.Array):
+                # xarray DataArray
+                lons = lons.data
+                lats = lats.data
+            elif not isinstance(lons, da.Array):
+                lons = da.from_array(np.asanyarray(lons), chunks=chunks)
+                lats = da.from_array(np.asanyarray(lats), chunks=chunks)
+        if data_slice is not None:
+            lons, lats = lons[data_slice], lats[data_slice]
+        return lons, lats
 
-    def get_lonlats_dask(self, chunks=CHUNK_SIZE):
+    def get_lonlats_dask(self, chunks=None):
```

Comment:
Why not leave the default instead of adding an if statement below ?

Reply:
No strong feelings, just less code lines. But you seem to have good reasons, so leave it be.